### PR TITLE
Create generic chargepoint

### PR DESCRIPTION
**Pull request by OrangeTux**

### CODE REVIEW

File: ocpp/charge_point.py [hunk 1]
```diff
@@ -257,15 +263,17 @@ def __init__(self, id, connection, response_timeout=30):
         if response.message_type_id == MessageType.CallError:
             LOGGER.warning("Received a CALLError: %s'", response)
             return
+        else:
+            response.action = call.action
+            validate_payload(response, self.ocpp_version)
```
Comment (tropxy) [inline ocpp/charge_point.py]: where is the self.ocpp_version set?

File: ocpp/messages.py [hunk 2]
```diff
@@ -50,50 +61,71 @@ def pack(msg):
...
-def get_schema(name):
+def get_schema(message_type_id, action, ocpp_version):
     """
     Read schema from disk and return in. Reads will be cached for performance
     reasons.
+
     """
-    if name in _schemas:
-        return _schemas[name]
+    if ocpp_version == "1.6":
```
Comment (tropxy) [inline ocpp/messages.py]: we could do some string operations here and substitute the if elses clauses by:

```python
schemas_dir = "v" + ocpp_version.replace(".", "")
```
so then if new versions come we dont need to do a train of if clauses
what do you think?

Comment (tropxy) [reply ocpp/messages.py]: for the error raise, maybe we can just check if the ocpp_version is in the list ["1.6", "2.0", ...]

File: ocpp/messages.py [hunk 3]
```diff
@@ -50,50 +61,71 @@ def pack(msg):
...
+        schema = get_schema(
+            message.message_type_id, message.action, ocpp_version
+        )
     except (OSError, json.JSONDecodeError) as e:
         raise ValidationError("Failed to load validation schema for action "
-                              f"'{action}': {e}")
+                              f"'{message.action}': {e}")
 
-    if action in ['SetChargingProfile', 'RemoteStartTransaction']:
+    if message.action in ['SetChargingProfile', 'RemoteStartTransaction']:
```
Comment (tropxy) [inline ocpp/messages.py]: for ocpp version 1.6 we have "RemoteStartTransaction", but for 2.0 we have the new "RequestStartTransaction". So, this option should be added

File: ocpp/v20/schemas/AuthorizeRequest_v1p0.json [hunk 4]
```diff
@@ -1,6 +1,5 @@
 ﻿{
```
Comment (tropxy) [inline ocpp/v20/schemas/AuthorizeRequest_v1p0.json]: Do you think there is a way we can inform why we have removed this id reference, so next time with even new schemas we know we have to remove it?

File: ocpp/charge_point.py (hunk 1 above)
Comment (OrangeTux) [reply ocpp/charge_point.py]: In `ocpp.{v16,v20}.ChargePoint`

File: ocpp/messages.py (hunk 2 above)
Comment (OrangeTux) [reply ocpp/messages.py]: Sure, I like it.

Comment (OrangeTux) [reply ocpp/messages.py]: Sure, I like it.

File: ocpp/v20/schemas/AuthorizeRequest_v1p0.json (hunk 4 above)
Comment (tropxy) [reply ocpp/v20/schemas/AuthorizeRequest_v1p0.json]: schade

**Pull request merged**